go-controller/pkg/ovn/ovn.go

```diff
@@ -403,41 +396,6 @@ func shouldUpdateNode(node, oldNode *kapi.Node) (bool, error) {
 	return true, nil
 }
 
-func newServiceController(client clientset.Interface, nbClient libovsdbclient.Client, recorder record.EventRecorder) (*svccontroller.Controller, informers.SharedInformerFactory, error) {
-	// Create our own informers to start compartmentalizing the code
-	// filter server side the things we don't care about
-	noProxyName, err := labels.NewRequirement("service.kubernetes.io/service-proxy-name", selection.DoesNotExist, nil)
-	if err != nil {
-		panic(err)
-	}
-
-	noHeadlessEndpoints, err := labels.NewRequirement(kapi.IsHeadlessService, selection.DoesNotExist, nil)
-	if err != nil {
-		panic(err)
-	}
-
-	labelSelector := labels.NewSelector()
-	labelSelector = labelSelector.Add(*noProxyName, *noHeadlessEndpoints)
```

Comment:
I dont see you porting this to the shared factory? Should we be implementing it there, or filter it out in the handler?

Reply:
so this code applies `noProxyName`, and `noHeadlessEndpoints` to everything (even though these labels only make sense for services and endpoint slices correspondingly), and watch factory that we use now applies 
- `noAlternateProxySelector` https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/factory/factory.go#L225 for Services (which is exactly the same as `noProxyName`) 
- `withServiceNameAndNoHeadlessServiceSelector` for EndpointSlices https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/factory/factory.go#L234C4-L234C47 which combines `noHeadlessEndpoints` filter with `LabelServiceName` requirement, which is used to associate services with endpointslices, and I think should always be present

So the filtering is done by the factory